### PR TITLE
Add min_zoom to templated SQL queries

### PR DESCRIPTION
**Pull request by iandees**
- [x] Update tests
- [x] Update data migrations
- [x] Update docs

Connects to #976.

### CODE REVIEW

File: queries/boundaries.jinja2 [hunk 1]
```diff
@@ -84,6 +86,7 @@ UNION ALL
 
 SELECT
   NULL AS name,
+  TODO_FIX_ME as min_zoom,
```
Comment (iandees) [inline queries/boundaries.jinja2]: Not sure what to use here for `buffered_land`.

File: queries/roads.jinja2 [hunk 2]
```diff
@@ -3,6 +3,7 @@ SELECT
     gid AS __id__,
     {% filter geometry %}the_geom{% endfilter %} AS __geometry__,
     'naturalearthdata.com' AS source,
+    TODO_FIX_ME as min_zoom,
```
Comment (iandees) [inline queries/roads.jinja2]: Not sure what to use here for `ne_10m_roads`.

File: queries/boundaries.jinja2 (hunk 1 above)
Comment (zerebubuth) [inline queries/boundaries.jinja2]: Because we include this at all zoom levels, does `0 AS min_zoom` make sense?

File: queries/roads.jinja2 (hunk 2 above)
Comment (zerebubuth) [inline queries/roads.jinja2]: We seem to include roads where `scalerank <= zoom`, so I think that effectively means `scalerank = min_zoom`.

File: queries/boundaries.jinja2 [hunk 3]
```diff
@@ -84,6 +86,7 @@ UNION ALL
 
 SELECT
   NULL AS name,
+  0 as min_zoom,
```
Comment (nvkelso) [inline queries/boundaries.jinja2]: Should this kick in at zoom 8, when we switch over from Natural Earth to OpenStreetMap?

**Comment by nvkelso** [comment]
Generally looks good to me. @zerebubuth for a once over.

**Comment by nvkelso** [comment]
@iandees Can you please add tests to verify `min_zoom` values are exporting in those 4 layers (there should be 8 tests: 4 low zoom Natural Earth tests and 4 OpenStreetMap mid-zoom or high zoom tests).

**Comment by iandees** [comment]
Building the fixtures for the tests here is taking a pretty long time. Still going…

**Comment by iandees** [comment]
@rmarianski, can you take a quick peek at this? The various pieces of this have been looked at, but a quick gander over the whole thing would be helpful.

**Comment by rmarianski** [comment]
:+1:

**Comment by iandees** [comment]
Thanks!

**Pull request merged**